core/src/main/java/org/bitcoinj/store/FlatDB.java

```diff
@@ -194,16 +194,39 @@ ReadResult read(Type object, boolean fDryRun) {
                 magicMessageTmp = new String(vchData, 0, magicMessage.length());
 
                 log.info("file magic message: {}",magicMessageTmp);
+                int fileVersion = 1;
+                try {
+                    String fileVersionString = magicMessageTmp.substring(magicMessageTmp.lastIndexOf('-') + 1);
+                    fileVersion = Integer.parseInt(fileVersionString);
+                } catch (NumberFormatException x) {
+                    //swallow
+                }
```

Comment:
The files dashj uses to store some structures for later have a file version which will allow future version to interpret them correctly.